python/manageJobs.py

```
@@ -260,7 +285,55 @@ def manageJobs(argv=None):
                 # actions that can be applied to all jobs
                 jobnums = [j.num for j in jobs]
                 schedd.act(htcondor.JobAction.Remove,jobnums)
+            elif options.xrootdResubmit:
+                fprint("")
+                file_and_site_per_file = {}
+                try:
+                    file_and_site_per_file = find_input_file_site_per_job(
+                        classad = options.inputFileClassAd,
+                        condor_jobs = jobs,
+                        log_key = options.logKey if options.logKey and options.logPath else "",
+                        log_path = options.logPath if options.logKey and options.logPath else "",
+                        prefer_us_sites = options.preferUSSites,
+                        verbose = options.verbose,
+                    )
+                except Exception as e:

```

why catch and return instead of letting execution just end from the original exception in file_finder?

---

I wanted it to be a little more graceful and I wanted to cleanup some of the other messages which get printed out when you raise an exception. For example:
```python
>>> raise Exception("stuff")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
Exception: stuff
```
I didn't want all of the `Traceback` printouts. I wanted it to look more like an OptionParser error.

---

Hiding exception tracebacks is an antipattern that makes it hard to debug.

If the only exceptions in file_finder are parser.error-like and you want that behavior, just print a message and then call `sys.exit(2)` instead of using an exception.

---

Fixed